src/Spectre.Data.Tests/Datasets/BasicTextDatasetTest.cs

```diff
@@ -0,0 +1,292 @@
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NUnit.Framework;
+using Spectre.Data.Datasets;
+using Spectre.Data.Structures;
+
+namespace Spectre.Data.Tests.Datasets
+{
+    [TestFixture, Category("Dataset")]
+    public class BasicTextDatasetTest
+    {
+        private IDataset _dataset;
+
+        [OneTimeSetUp]
+        public void SetUpFixture()
+        {
+            Directory.SetCurrentDirectory(TestContext.CurrentContext.TestDirectory + "\\..\\..");
+        }
+
+        [SetUp]
+        public void SetUpClass()
+        {
+            double[] mz = {1.0, 2.0, 3.0};
+            double[,] data = {{1, 2.1, 3.2}, {4, 5.1, 6.2}, {7, 8.1, 9.2}};
+
+            _dataset = new BasicTextDataset(mz, data);
+        }
+
+        [Test]
+        public void CreateFromFileTest()
+        {
+            try
+            {
+                _dataset = new BasicTextDataset("small-test.txt");
+            }
+            catch (Exception e)
+            {
+                Assert.Fail("Dataset failed to initialize from file.", e);
+            }
+
+            IEnumerable<SpatialCoordinates> coordinates = _dataset.SpacialCoordinates;
+            var enumerator = coordinates.GetEnumerator();
+            enumerator.MoveNext();
+            SpatialCoordinates sc = enumerator.Current;
+            Assert.AreEqual(1, sc.X, "Spatial coordinates differ");
+            Assert.AreEqual(1, sc.Y, "Spatial coordinates differ");
+            Assert.AreEqual(0, sc.Z, "Spatial coordinates differ");
+
+            enumerator.MoveNext();
+            sc = enumerator.Current;
+            Assert.AreEqual(2, sc.X, "Spatial coordinates differ in second spectrum");
+            Assert.AreEqual(1, sc.Y, "Spatial coordinates differ in second spectrum");
+            Assert.AreEqual(0, sc.Z, "Spatial coordinates differ in second spectrum");
+
+            Assert.AreEqual(_dataset.GetRawMzArray(), new[] {899.99, 902.58, 912.04}, "The m/zs differ");
+            Assert.AreEqual(_dataset.GetRawIntensityArray(0), new[] {12.0, 20.0, 0.0},
+                "The intensities of first spectrum differs");
+        }
+
+        [Test]
+        public void CreateFromRawDataTest()
+        {
+            double[] mz = null;
+            double[,] data = null;
+
+            try
+            {
+                _dataset = new BasicTextDataset(mz, data);
+                Assert.Fail("Dataset accepted null raw data.");
```

Comment:
Such a check should be done separately - no one knows, whether it fails due to null mz, data or both.